### PR TITLE
[HiTL] Revamped annotation tool pipeline with more robustness & parameteriztion

### PR DESCRIPTION
**Pull request by snyxan**
# Description

This PR revamped part of the annotation tools we already have, fixed a bunch of bugs related to HIT/turk and added necessary pieces in order to be used by HiTL pipeline.

## Type of change

Please check the options that are relevant.

- [x] Bug fix (non-breaking change that fixes an issue)
- [x] Proposes a change (non-breaking change that isn't necessarily a bug)
- [ ] Refactor
- [ ] New feature (non-breaking change that adds a new functionality)
- [ ] Breaking change (fix or feature that would break some existing functionality downstream)
- [ ] This is a unit test
- [ ] Documentation only change
- [ ] Datasets Release
- [ ] Models Release

## Type of requested review

- [x] I want a thorough review of the implementation.
- [x] I want a high level review. 
- [ ] I want a deep design review.

## Before and After

**Before**

Even using the annotation tool alone is error-prune. Mainly because it track HIT status incorrectly. The problems are:

1. it doesn't track the HIT which is running the annotation tool. It only check if there's any HITs reviewable of this account and retrieve all the data from it. This is problematic because multiple different kinds of HITs could be run within the same account and you must explicitly query the HIT that corresponded to your annotation job.

2. The waiting mechanism is error-prune. It only waits for a fixed time period and then proceed. Ideally it should keep checking the status of the HIT you are interested in and get the data once it's ready (completed by turker).

**After**

The tool can explicitly query HITs that corresponded to your interaction job, waiting for it to complete and then get the result from it. Also added a timeout for it to avoid indefinitely run.

# Testing

Tested and passed

# Checklist:

- [x] I have performed manual end-to-end testing of the feature in my environment.
- [x] I have added Docstrings and comments to the code.
- [ ] I have made changes to existing documentation where needed.
- [ ] I have added tests that show that the PR is functional.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have added relevant collaborators to review the PR before merge.
- [ ] [Polymetis only] I have verified that all scripts in `tests/scripts` runs on hardware.

### CODE REVIEW

File: tools/annotation_tools/turk_with_s3/get_results.py [hunk 1]
```diff
@@ -35,6 +43,92 @@ def get_hit_list_status(mturk):
...
+                res = res.append(new_row, ignore_index=True)
+                res.to_csv(output_csv, index=False)
+
+            mturk.approve_assignment(AssignmentId=assignment["AssignmentId"])
+            mturk.delete_hit(HITId=hit_id)
+    else:
+        logging.info("No results ready yet")
+        # if returned assignment is empty,reject
+        mturk.delete_hit(HITId=hit_id)
+
```
Comment (snyxan) [inline tools/annotation_tools/turk_with_s3/get_results.py:129]: @kavyasrinet there're a lot of duplicate code due to the similarities between tool A/B/C/D. You should focus on this function.

File: tools/annotation_tools/turk_with_s3/run_all_tasks.py [hunk 2]
```diff
@@ -12,14 +14,19 @@
...
-default_write_dir = os.path.dirname(os.path.abspath(__file__))
+parser = argparse.ArgumentParser()
+parser.add_argument("--default_write_dir", type=str, required=True)
+parser.add_argument("--timeout", type=float, default=60)
+parser.add_argument("--dev", default=False, action="store_true")
+
+args = parser.parse_args()
+dev_flag = "--dev" if args.dev else ""
+default_write_dir = args.default_write_dir
+timeout = 300#args.timeout
```
Comment (kavyasrinet) [inline tools/annotation_tools/turk_with_s3/run_all_tasks.py:26]: nit: remove comment

File: tools/annotation_tools/turk_with_s3/run_tool_1A.py [hunk 3]
```diff
@@ -16,69 +22,105 @@
...
+parser.add_argument("--default_write_dir", type=str, required=True)
+parser.add_argument("--dev", default=False, action="store_true")
+parser.add_argument("--timeout", type=float, default=60)
+
+args = parser.parse_args()
+dev_flag = "--dev" if args.dev else ""
+default_write_dir = args.default_write_dir
+timeout = args.timeout
+
+# dev_flag = ""
```
Comment (kavyasrinet) [inline tools/annotation_tools/turk_with_s3/run_tool_1A.py:35]: nit: remove comments

File: tools/annotation_tools/turk_with_s3/run_tool_1A.py [hunk 4]
```diff
@@ -16,69 +22,105 @@
...
+# # Parse flags passed into script run
+# if len(sys.argv) > 1:
+#     # flag to toggle dev mode is --dev
+#     dev_flag = sys.argv[1]
 
 # CSV input
 rc = subprocess.call(
     [
-        "python3 construct_input_for_turk.py --input_file input.txt > A/turk_input.csv"
+        # "python3 ../text_to_tree_tool/construct_input_for_turk.py --input_file input.txt > A/turk_input.csv"
```
Comment (kavyasrinet) [inline tools/annotation_tools/turk_with_s3/run_tool_1A.py:44]: same

File: tools/annotation_tools/turk_with_s3/run_tool_1A.py [hunk 5]
```diff
@@ -16,69 +22,105 @@
...
+    aws_secret_access_key=secret_key,
+    region_name=aws_region,
+    endpoint_url=MTURK_URL,
 )
-if rc != 0:
-    print("Error fetching HIT results. Exiting.")
-    sys.exit()
+
+get_hit_result(mturk, hit_id, f"{default_write_dir}/A/turk_output.csv", True if dev_flag else False, timeout)
+
```
Comment (kavyasrinet) [inline tools/annotation_tools/turk_with_s3/run_tool_1A.py:89]: nit: remove

File: tools/annotation_tools/turk_with_s3/run_tool_1D.py [hunk 6]
```diff
@@ -68,7 +107,7 @@
 # Postprocess
 print("*"*50)
 print("*** Postprocessing results ***")
-rc = subprocess.call(["python3 parse_tool_D_outputs.py"], shell=True)
+rc = subprocess.call([f"python3 parse_tool_D_outputs.py --folder_name {default_write_dir}/D/"], shell=True)
 if rc != 0:
     print("Error collating answers. Exiting.")
     sys.exit()
```
Comment (kavyasrinet) [inline tools/annotation_tools/turk_with_s3/run_tool_1D.py:113]: Thanks for cleaning these up, we could even rewrite the subprocess calls to files like: `parse_tool_D_outputs.py`

This version definitely needed rewriting. The work I had done here was to just make sure the pre existing version could work with Turk.

**Review by kavyasrinet — CHANGES_REQUESTED** [review]

File: tools/annotation_tools/turk_with_s3/run_tool_1D.py (hunk 6 above)
Comment (snyxan) [reply tools/annotation_tools/turk_with_s3/run_tool_1D.py]: created an issue for that #804. We can tackle it when have free cycle.

**Review by kavyasrinet — APPROVED** [review]

**Pull request merged**